File: DSA-DAILY/DSA-DAILY/30-07-2025/2419. Longest Subarray With Maximum Bitwise AND.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans=0,maxs=0,s=0;
        for(int num:nums){
            if(num>maxs){
                maxs=num;
                ans=s=0;
            }
            if(maxs==num){
                s++;
            }
            else{
                s=0;
            }
            ans=max(ans,s);
        }
        return ans;
    }
};

int main(){
    Solution sol;
    int n;
    cout << "Enter size of array: ";
    cin >> n;
    cout << "Enter Array: ";
    vector<int> a(n);
    for(int i = 0; i < n; i++){
        cin >> a[i];
    }
    int result = sol.longestSubarray(a);
    cout << "Length of longest subarray with maximum bitwise AND: " << result << endl;
    return 0;
}

File: DSA-DAILY/30-07-2025/2419. Longest Subarray With Maximum Bitwise AND.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
    let ans=0,s=0,maxi=0;
    for(let i of nums){
        if(i>maxi){
            maxi=i;
            s=ans=0;
        }
        if(maxi==i){
            s++;
        }
        else{
            s=0;
        }
        if(ans<s)ans=s;
    }
    return ans;
};
// Node.js input handling
const readline = require('readline');   
const rl = readline.createInterface({
    input: process.stdin,   
    output: process.stdout
});
rl.question('Enter array elements separated by space: ', (input) => {
    const nums = input.trim().split(/\s+/).map(Number);
    const result = longestSubarray(nums);
    console.log('Longest subarray with maximum bitwise AND:', result);
    rl.close();
});
